### PR TITLE
Add RelativeBasePath parameter to Resolve-Path

### LINKED ISSUE #8435: Proposal: Resolve-Path with -Relative parameter should allow the use of a reference path
# Summary of the new feature/enhancement

As a PowerShell developer, I want to be able to obtain the path of an item relative to a path that is not my current directory so that I can avoid changing my current location context for a relative directory lookup. 

# Proposed technical implementation details

Assuming I have a directory like this:

```none
/MyProject
    |-/dir
        |-test_1.txt
        |-test_2.txt
        |-/subdir
            |-test_3.txt
```

Currently, if I want to do a resolve a path relative to a *different* directory, I have to use set-location before calling `Resolve-Path`. For example, if I wanted to get all my `.txt` file locations *relative to `dir`*, I would have to do something like this:

```powershell
/MyProject> cd dir
/MyProject/dir> Get-ChildItem -Recurse -Include "*.txt" | Resolve-Path -Relative
./subdir/test_3.txt
./test_1.txt
./test_2.txt
/MyProject/dir> cd ..
```

Ideally, I would like to do something like this:

```powershell
/MyProject> Get-ChildItem .\dir -Recurse -Include "*.txt" | Resolve-Path -Relative -ReferencePath .\dir
./subdir/test_3.txt
./test_1.txt
./test_2.txt
```

### PR DESCRIPTION
**Pull request by MartinGC94**
# PR Summary
Adds a `RelativeBasePath` path parameter to `Resolve-Path` which allows the user to specify a path to base the relative path off.

For example, if the current location is C:\Windows\System32 and the user wants to resolve the relative path to the temp folder from their home folder they can do it like this: Resolve-Path -Path $env:TEMP -RelativeBasePath $HOME which would output: .\AppData\Local\Temp  

## PR Context
Fixes https://github.com/PowerShell/PowerShell/issues/8435

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [X] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [X] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/9940 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [X] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ResolvePathCommand.cs [hunk 1]
```diff
@@ -75,6 +75,12 @@ public SwitchParameter Relative
 
         private SwitchParameter _relative;
 
+        /// <summary>
+        /// Gets or sets the path the resolved relative path should be based off.
+        /// </summary>
+        [Parameter()]
+        public string RelativeBasePath { get; set; }
```
[the remark this answers is not in the document]
Comment (MartinGC94) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/ResolvePathCommand.cs]: Sure. This is what the parametersets look like now:
image

File: test/powershell/Modules/Microsoft.PowerShell.Management/Resolve-Path.Tests.ps1 [hunk 2]
```diff
@@ -48,4 +48,8 @@ Describe "Resolve-Path returns proper path" -Tag "CI" {
             Pop-Location
         }
     }
+    It 'Resolve-Path -Relative should support user specified base paths' {
+        $Expected = Join-Path -Path .\ -ChildPath fakeroot
+        Resolve-Path -Path $fakeRoot -Relative -RelativeBasePath $testRoot | Should -BeExactly $Expected
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Resolve-Path.Tests.ps1:53]: Shouldn't using both `-Relative` and `-RelativeBasePath` now be invalid?

Comment (MartinGC94) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Resolve-Path.Tests.ps1]: Yes. I forgot to change the test.

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

**Pull request merged**